packages/test/test-end-to-end-tests/src/test/gc/gcUnreferencePhases.spec.ts

```diff
@@ -0,0 +1,133 @@
+        await waitForContainerConnection(mainContainer);
+
+        const {
+            container: summarizingContainer,
+            summarizer,
+        } = await loadSummarizerAndContainer();
+
+        // create datastore and blob
+        const dataStore = await mainDataStore._context.containerRuntime.createDataStore(TestDataObjectType);
+        const dataStoreHandle = dataStore.entryPoint;
+        assert(dataStoreHandle !== undefined, "Expected a handle when creating a datastore");
+        const blobContents = "Blob contents";
+        const blobHandle = await mainDataStore._runtime.uploadBlob(stringToBuffer(blobContents, "utf-8"));
+
+        // store datastore and blob handles
+        mainDataStore._root.set("dataStore", dataStoreHandle);
+        mainDataStore._root.set("blob", blobHandle);
+
+        // unreference datastore and blob handles
+        mainDataStore._root.delete("dataStore");
+        mainDataStore._root.delete("blob");
+
+        // Summarize and verify datastore and blob are unreferenced and not tombstoned
+        await provider.ensureSynchronized();
+        let summaryTree = (await summarizeNow(summarizer)).summaryTree;
+        const gcState = getGCStateFromSummary(summaryTree);
+        assert(gcState !== undefined, "Expected GC state to be generated");
+        assert(gcState.gcNodes[dataStoreHandle.absolutePath] !== undefined, "Datastore should exist on gc graph");
+        assert(gcState.gcNodes[dataStoreHandle.absolutePath].unreferencedTimestampMs !== undefined, "Datastore should be unreferenced");
+        assert(gcState.gcNodes[blobHandle.absolutePath] !== undefined, "Blob should exist on gc graph");
+        assert(gcState.gcNodes[blobHandle.absolutePath].unreferencedTimestampMs !== undefined, "Blob should be unreferenced");
+        let tombstoneState = getGCTombstoneStateFromSummary(summaryTree);
+        assert(tombstoneState === undefined, "Nothing should be tombstoned");
+
+        // Wait inactive timeout
+        await delay(inactiveTimeoutMs);
+        // Summarize and verify datastore and blob are unreferenced and not tombstoned
+        // Functionally being inactive should have no effect on datastores
+        mainDataStore._root.set("send", "op");
+        await provider.ensureSynchronized();
+        summaryTree = (await summarizeNow(summarizer)).summaryTree;
+        assert(summaryTree.tree[gcTreeKey].type === SummaryType.Handle, "GC tree should not have changed");
+
+        // Wait sweep timeout
+        await delay(sweepTimeoutMs);
+        mainDataStore._root.set("send", "op2");
+        await provider.ensureSynchronized();
+        summaryTree = (await summarizeNow(summarizer)).summaryTree;
+        const rootGCTree = summaryTree.tree[gcTreeKey];
+        assert (rootGCTree !== undefined, "GC tree should be a tree");
+        assert(rootGCTree.type === SummaryType.Tree, `GC data should be a tree`);
```

Comment:
nit: You could do these two in one assert - `assert(rootGCTree?.type === SummaryTree.Tree`